R/network.R

```diff
@@ -6,46 +6,69 @@
 
 #' Transform parsed dependencies into a sdsmj file.
 #'
-#' # Creates a sdsm.json from a table of dependencies from Depends.
+#' Converts table of dependencies from \code{\link{parse_dependencies}} into an *-sdsm.json.
+#' In the sdsm.json, the Variables are all files/methods or any variables under analysis
+#' (rows/columns in dependency matrix) and the Cells (matrix cell) contain all the relations of
+#'  variable (src & dest) pairs.
 #'
-#' @param depends_table A parsed depends project by \code{\link{parse_dependencies}}.
-#' @param is_sorted whether to sort the variables (filenames) in the dsm.json files
+#' @param project_depends A parsed depends project by \code{\link{parse_dependencies}}.
+#' @param sdsmj_path the path to save the structural dsm (*-sdsm.json).
+#' @param is_directed whether the graph is directed. If false, cells will be doubled
+#'  with src/dest switched to represent an undirected graph as a directed graph.
+#' @param is_sorted whether to sort the variables (filenames) in the sdsm.json file.
 #' @export
 #' @family edgelists
 #' @family dv8
-#' @seealso \code{\link{parse_dependencies}}, \code{\link{transform_gitlog_to_bipartite_network}},
-#' \code{\link{graph_to_dsmj}}
-transform_dependencies_to_sdsmj <- function(depends_table, is_sorted=FALSE){
+#' @seealso \code{\link{parse_dependencies}} to get a table of parsed dependencies needed
+#'  as input into \code{\link{transform_dependencies_to_sdsmj}},
+#'  \code{\link{transform_gitlog_to_hdsmj}} to perform a similar transformation into a *-dsm.json using a gitlog,
+#' \code{\link{graph_to_dsmj}} to generate a *-dsm.json file. A *-sdsm.json will be returned for \code{\link{transform_dependencies_to_sdsmj}}
+#' and a *-hdsm.json will be returned for \code{\link{transform_gitlog_to_hdsmj}}
+transform_dependencies_to_sdsmj <- function(project_depends, sdsmj_path, is_directed, is_sorted=FALSE){
   # Rename "filepath" to "to"
-  colnames(depends_table$nodes)[1] = "name"
+  colnames(depends_table$nodes)[colnames(depends_table$nodes) == "filepath"] = "name"
   # Rename "src_filepath" to "from"
-  colnames(depends_table$edgelist)[1] = "from"
+  colnames(depends_table$edgelist)[colnames(depends_table$edgelist) == "src_filepath"] = "from"
   # Rename "dest_filepath" to "to"
-  colnames(depends_table$edgelist)[2] = "to"
+  colnames(depends_table$edgelist)[colnames(depends_table$edgelist) == "dest_filepath"] = "to"
```

Comment:
Use:

setnames(x=depends_table[["edgelist"]],
                  old = c("src_filepath","dest_filepath") 
                  to = c("from","to")                             
                  )
This should not require an assignment. It is one of the peculiarities of data.table, and why it makes it so fast.